projects/packages/forms/src/blocks/form-step/edit.js

```diff
@@ -122,6 +122,7 @@ function StepBreak( { stepLabel, currentIndex, setAttributes, clientId } ) {
 		<div className="jetpack-form-step__break">
 			{ /* Allow inline editing of step label */ }
 			<RichText
+				allowedFormats={ [] }
```

Comment:
I tried also `withoutInteractiveFormatting` but that didn't work for some reason. I don't see why not. :-/ 

https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/rich-text/README.md#withoutinteractiveformatting-boolean

It might also make sense to just use `input` here: https://developer.wordpress.org/block-editor/reference-guides/richtext/#unwanted-formatting-options-still-display